_docker/drupal/drupal-filesystem/web/modules/custom/rhd_purpose_attributes/src/Controller/PurposeAttributesPageController.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Drupal\rhd_purpose_attributes\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Link;
+use Drupal\core\Url;
+
+/**
+ * Class PurposeAttributesPageController.
+ *
+ * @package Drupal\rhd_purpose_attributes\Controller
+ */
+class PurposeAttributesPageController extends ControllerBase {
+
+  /**
+   * Returns a purpose_attributes_report_block block.
+   *
+   * @return array
+   *   Returns a block render array.
+   */
+  public function renderBlockContent() {
+    $plugin_block = \Drupal::service('plugin.manager.block')
+      ->createInstance('purpose_attributes_report_block', []);
+    $access_result = $plugin_block->access(\Drupal::currentUser());
```

Comment:
I believe that you can get the current user from the ControllerBase https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Controller%21ControllerBase.php/function/ControllerBase%3A%3AcurrentUser/8.2.x